### ISSUE TITLE
One GLA Battle Bus passenger automatically dies when host vehicle is sniped by GLA Jarmen Kell

### ISSUE THREAD

**Issue by xezon** [TheSuperHackers/GeneralsGamePatch#578]
One GLA Battle Bus passenger automatically dies when host vehicle is sniped by GLA Jarmen Kell. Reported by Dominator.

**Comment by xezon** [comment]
I think the source problem here is that neutral vehicles can crush infantry. When you snipe any transport vehicle, it will have a little bit of velocity left while the passengers are forced to leave. Since vehicle and passenger are no longer in the same team, the vehicle will be able to kill them when moved.

This issue can also be observed on USA Humvee and China Outpost.

**Comment by EnigmaZift** [comment]
Have you tried while the unit is stationary?

**Comment by commy2** [comment]
Jarmen Kell can also immediately shoot infantry 2 frames after Sniper Attack, so be careful when reproducing this.

**Comment by EnigmaZift** [comment]
> Jarmen Kell can also immediately shoot infantry 2 frames after Sniper Attack, so be careful when reproducing this.

I can confirm this, since his weapons are not linked and they have 0 pre-fire delay. So actually he should be able to do so literally the next logic frame (2 engine frames) (66 ms)

**Comment by MTKing4** [comment]
Found an instance of this happening on  stream, watch from 14:30 from then you'll see that the Battle bus has jarmen as the first unit, few seconds later bus gets sniped and you hear the jarmen death sound (Indicating he was killed not crushed) with the same shot
https://youtu.be/mFF3fH7TLIs

**Comment by xezon** [comment]
Looks like commy2 suspicion is correct. So fix would need to increase Jarmen cooldown after snipe if possible.
https://youtu.be/mFF3fH7TLIs?t=880

**Comment by commy2** [comment]
I am not denying the neutral vehicle crushing the first passenger theory. Both could be separate issues.

**Comment by MTKing4** [comment]
Crushing scenario is possible as well i think yeah, but in this case commando units are not crushable anyway so yeah

**Comment by EnigmaZift** [comment]
Two things:

1. The infantry inside the Battle Bus is not getting crushed. I say this with a decent level of certainty. It's impossible for Jarmen to even get crushed by any Battle Bus, neutral, allied or enemy. Therefore the only correct conclusion is that the game simply decides to kill the first passenger. 
This has nothing to do with the snipe, it's probably hardcoded, and also probably intentional.

2. You cannot add cooldown to the main gun after the pilot kill ability without introducing issues. The best way around this is to give the main gun a "pre-fire delay" per shot. To retain balance basically, we might remove a couple frames from his reload time and attach it to his pre-fire delay. A delay like this can be known as an aiming delay, and basically he will have to stay still for a couple milliseconds before making his shot. The issue from this is of course, his shots will not be instant and he might feel a little off. Rest assured though, the DPS will stay the same.

**Comment by commy2** [comment]
> The infantry inside the Battle Bus is not getting crushed. It's impossible for Jarmen to even get crushed by any Battle Bus[.]

There can be other infantry inside the Bus, or Humvee, or Bunkerlord besides a Jarmen.

> Therefore the only correct conclusion is that the game simply decides to kill the first passenger.

This is not my conclusion at all.

> The best way around this is to give the main gun a "pre-fire delay" per shot. To retain balance basically, we might remove a couple frames from his reload time and attach it to his pre-fire delay.

Bad idea. Completely changes Jarmen v Jarmen meta.

**Comment by MTKing4** [comment]
> Therefore the only correct conclusion is that the game simply decides to kill the first passenger

If you look closely you'll see his gun flashing twice so it's highly likely a second shot

**Comment by xezon** [comment]
> I am not denying the neutral vehicle crushing the first passenger theory. Both could be separate issues.

Yeah that is certainly an issue. I have seen it plenty times before. We can track it separately.

* #585
